### ISSUE TITLE
Add `dedicationOfAParticularChurch` to `general.js`

### ISSUE THREAD

**Issue by tukusejssirs** [romcal/romcal#136]
`dedicationOfAParticularChurch` celebration definition should be moved from particular calendars to `general.js`. It is celebrated on 26 Oct as solemnity in churches of which the date of consecration is unknown.

**Comment by emagnier** [comment]
I'm not sure about moving to general, because this solemnity will overwrite the feria or any other celebration on this day. And this might not be relevant for all church.
And actually, "optional solemnity" doesn't exist in romcal.

**Comment by tukusejssirs** [comment]
After searching a bit on the Internet, you might be right. Even with your implicit suggestion to remove this celebration from `romcal` (non-parish calendars) altogether. Then, in a separate repo, we could create the modification of `romcal` for _local church calendars_ (i.e. the diocesan and parish calendars).

[Wiki](https://en.wikipedia.org/wiki/General_Roman_Calendar) has several celebration definitions in different particular or religious calendars (and on different dates). It even adds the [following](https://en.wikipedia.org/wiki/General_Roman_Calendar#Local_calendars):

> The calendar for a diocese is typically based on a national calendar, such as those listed above, with a few additions. For instance, the anniversary of the dedication of the cathedral is celebrated as a Solemnity in the cathedral church and as a Feast in all the other churches of the diocese. The feast day of the principal Patron saint of the diocese is celebrated as a Feast throughout the diocese. ( [General Norms for the Liturgical Year and the Calendar](http://www.catholicculture.org/library/view.cfm?recnum=5932), Table of Liturgical Days according to their order of precedence, 4 and 8.)
>
> The calendar of a parish is based on the calendar of its diocese, but—in addition to the celebrations in the diocesan calendar—there are other celebrations, including the anniversary of the dedication of the parish church and the feast day of the principal Patron saint of the church, both of which are celebrated as Solemnities.

**Comment by tukusejssirs** [comment]
It has just occured to me that we could add a romcal option (like `addDedicationOfAParticularChurch`) that would add the celebration to the output if set to `true` (else it would not output it; `false` by default), because many Ordos (Slovakia, Poland, Canada, etc) are including it. IMO if we want to implement #20, we should want to have all options available in order to be as faithfull as possible.

Note to myself: when it’s gonna be implemented, `dedication_of_consecrated_churches` is to be celebrated on the following dates in the following calendars:
- `canada.ts`: 22 October ([src](https://www.cccb.ca/wp-content/uploads/2017/09/nationallitcalender2016_01_27.pdf));
- `czechRepublic.ts`: 25 October or the following Sunday;
- `poland.ts`: last Sunday of October;
- `slovakia.ts`: 26 October.

**Comment by tukusejssirs** [comment]
I still think that using a romcal option (e.g. `addDedicationOfAParticularChurch`) is the best idea, as it is usually not celebrated at all.

On the other hand, we should merge all `dedication_of_consecrated_churches*` keys into one `dedication_of_consecrated_churches` key, as there is only one such celebration in a particular calendar.

**Comment by tukusejssirs** [comment]
I have just noticed that #431 introduced a potential anomaly.

`es` and `pt-br` are the only locale that has `dedication_of_consecrated_churches_on_last_sunday_of_october` and `dedication_of_consecrated_churches_on_october_25` keys localised.

```bash
# grep -r dedication_of_consecrated_churches lib/
lib/general-calendar/proper-of-saints.ts:    dedication_of_consecrated_churches_on_october_25: {
lib/general-calendar/proper-of-saints.ts:      customLocaleId: 'dedication_of_consecrated_churches',
lib/general-calendar/proper-of-saints.ts:    dedication_of_consecrated_churches_on_last_sunday_of_october: {
lib/general-calendar/proper-of-saints.ts:      customLocaleId: 'dedication_of_consecrated_churches',
lib/locales/la.ts:    dedication_of_consecrated_churches: 'In Dedicatione Ecclesiæ, cuius dies consecrationis ignotus est',
lib/locales/pl.ts:    dedication_of_consecrated_churches: 'Rocznica poświęcenia kościoła własnego',
lib/locales/cs.ts:    dedication_of_consecrated_churches: 'Výročí posvěcení kostela',
lib/locales/es.ts:    dedication_of_consecrated_churches_on_last_sunday_of_october:
lib/locales/es.ts:    dedication_of_consecrated_churches_on_october_25: 'Dedicación de iglesias consagradas el 25 de octubre',
lib/locales/sk.ts:    dedication_of_consecrated_churches: 'Výročie posviacky chrámov, ktorých deň posviacky nie je známy',
lib/locales/en.ts:    dedication_of_consecrated_churches: 'The Dedication of Consecrated Churches Whose Date of Consecration is Unknown',
lib/locales/pt-br.ts:    dedication_of_consecrated_churches_on_last_sunday_of_october: 'Dedicação de Igrejas Consagradas no Último Domingo de Outubro',
lib/locales/pt-br.ts:    dedication_of_consecrated_churches_on_october_25: 'Dedicação de Igrejas Consagradas em 25 de Outubro',
lib/particular-calendars/slovakia.ts:    dedication_of_consecrated_churches_on_october_25: {
lib/particular-calendars/slovakia.ts:    dedication_of_consecrated_churches_on_last_sunday_of_october: {
lib/particular-calendars/poland.ts:    dedication_of_consecrated_churches_on_october_25: {
lib/particular-calendars/canada.ts:    dedication_of_consecrated_churches_on_october_25: {
lib/particular-calendars/canada.ts:    dedication_of_consecrated_churches_on_last_sunday_of_october: {
lib/catalog/martyrology.ts:    dedication_of_consecrated_churches: {
```

While those keys are used in some calendars (`canada`, `poland`, `slovakia`), IMHO they are used incorrectly.

```bash
grep -r dedication_of_consecrated_churches lib/particular-calendars/ -A2
lib/particular-calendars/slovakia.ts:    dedication_of_consecrated_churches_on_october_25: {
lib/particular-calendars/slovakia.ts-      dateDef: { month: 10, date: 26 },
lib/particular-calendars/slovakia.ts-    },
--
lib/particular-calendars/slovakia.ts:    dedication_of_consecrated_churches_on_last_sunday_of_october: {
lib/particular-calendars/slovakia.ts-      drop: true,
lib/particular-calendars/slovakia.ts-    },
--
lib/particular-calendars/poland.ts:    dedication_of_consecrated_churches_on_october_25: {
lib/particular-calendars/poland.ts-      drop: true,
lib/particular-calendars/poland.ts-    },
--
lib/particular-calendars/canada.ts:    dedication_of_consecrated_churches_on_october_25: {
lib/particular-calendars/canada.ts-      dateDef: { month: 10, date: 22 },
lib/particular-calendars/canada.ts-    },
--
lib/particular-calendars/canada.ts:    dedication_of_consecrated_churches_on_last_sunday_of_october: {
lib/particular-calendars/canada.ts-      drop: true,
lib/particular-calendars/canada.ts-    },
```

IMHO the correct way to use these keys is the way it is done in `proper-of-saints.ts`:

```bash
$ grep -r dedication_of_consecrated_churches lib/general-calendar/ -A3
lib/general-calendar/proper-of-saints.ts:    dedication_of_consecrated_churches_on_october_25: {
lib/general-calendar/proper-of-saints.ts-      precedence: Precedences.ProperSolemnity_DedicationOfTheOwnChurch_4b,
lib/general-calendar/proper-of-saints.ts:      customLocaleId: 'dedication_of_consecrated_churches',
lib/general-calendar/proper-of-saints.ts-      dateDef: { month: 10, date: 25 },
lib/general-calendar/proper-of-saints.ts-      isOptional: true,
lib/general-calendar/proper-of-saints.ts-    },
--
lib/general-calendar/proper-of-saints.ts:    dedication_of_consecrated_churches_on_last_sunday_of_october: {
lib/general-calendar/proper-of-saints.ts-      precedence: Precedences.ProperSolemnity_DedicationOfTheOwnChurch_4b,
lib/general-calendar/proper-of-saints.ts:      customLocaleId: 'dedication_of_consecrated_churches',
lib/general-calendar/proper-of-saints.ts-      dateDef: { month: 10, lastDayOfWeekInMonth: 0 },
lib/general-calendar/proper-of-saints.ts-      isOptional: true,
lib/general-calendar/proper-of-saints.ts-    },
```

However, IMHO this celebration should not be defined in the GRC and therefore they should be removed. Then, `dedication_of_consecrated_churches_on_*` would not be needed to be dropped in the particular calendars. 

WDYT, @emagnier? Should we remove the celebration from the GRC? Should we define the celebration in the particular calendars using `customLocaleId`? Finally, should we remove `dedication_of_consecrated_churches_on_*` l10n keys from `es` and `pt-br`?

I can do it, I just want to hear you opinion first. :pray:

**Comment by emagnier** [comment]
These two elements `dedication_of_consecrated_churches_on_october_25` and `dedication_of_consecrated_churches_on_last_sunday_of_october` are included in the Missal and are part of the general Roman calendar; therefore, their definitions should remain in the calendar. They are actually the same thing, but since they can be optionally placed on 2 possible dates, they need to be defined twice due to technical limitations in Romcal.

The decision to keep either of them, or to entirely remove them, lies with the specific calendars of churches, parishes, or dioceses. If a dedication of a consecrated church is known for a particular church and is placed on a different day, it's up to their discretion. In this case, in the church calendar, they should remove these 2 elements (using `drop: true`), and create a new definition for this day.

Furthermore, both of these definitions have a `customLocaleId` property, and the value is the same for both. This value (`dedication_of_consecrated_churches`) corresponds to the translation key to be used by the translation tool (i18next in our case). When the `customLocaleId` property is not specified, the key of the calendar item is used instead (which is the case most of the time).

**TL;DR 1:** `dedication_of_consecrated_churches_on_october_25` and `dedication_of_consecrated_churches_on_last_sunday_of_october` will never be used by the translation tool because both of these definitions have a `customLocaleId` property specifying a different translation key.

**TL;DR 2:**
- `dedication_of_consecrated_churches_on_october_25` and `dedication_of_consecrated_churches_on_last_sunday_of_october` should be removed from the localization files, but kept in the general Roman calendar file.
- `dedication_of_consecrated_churches` should be kept on the localization files.

**Comment by tukusejssirs** [comment]
> These two elements `dedication_of_consecrated_churches_on_october_25` and `dedication_of_consecrated_churches_on_last_sunday_of_october` are included in the Missal and are part of the general Roman calendar; therefore, their definitions should remain in the calendar.

Which Roman Missal you refer to? AFAIK:

- yes, they are part of the Roman Missal in vernacular;
- no, they aren’t part of the Latin Roman Missal:
  - even if I am wrong and they are listed in the Latin Roman Missal, they are not part of the GRC;
  - as per [this comment of mine](https://github.com/romcal/romcal/issues/136#issuecomment-1248397564), I think they should be part only of (generic) national/country calendars and lower calendars (diocesan/parish/church calendars) (when the exact dedication date is unknown), but not of the GRC.

Anyway, If possible, I’d like to see the source where they are _both_ listed in one calendar. :pray:

> They are actually the same thing, but since they can be optionally placed on 2 possible dates, they need to be defined twice due to technical limitations in Romcal.

Okay, even if we keep the celebrations defined as they are, IMHO we should have a single l10n key which would be the value of `customLocaleId` of those celebrations.

> The decision to keep either of them, or to entirely remove them, lies with the specific calendars of churches, parishes, or dioceses.

Okay, but have you encountered a particular calendar with more than one date for the unkown church dedications? (Known church dedication dates are always preferred AFAIK.)

> Furthermore, both of these definitions have a `customLocaleId` property, and the value is the same for both. This value (`dedication_of_consecrated_churches`) corresponds to the translation key to be used by the translation tool (i18next in our case). When the `customLocaleId` property is not specified, the key of the calendar item is used instead (which is the case most of the time).
>
> **TL;DR 1:** `dedication_of_consecrated_churches_on_october_25` and `dedication_of_consecrated_churches_on_last_sunday_of_october` will never be used by the translation tool because both of these definitions have a `customLocaleId` property specifying a different translation key.
>
> **TL;DR 2:**
> - `dedication_of_consecrated_churches_on_october_25` and `dedication_of_consecrated_churches_on_last_sunday_of_october` should be removed from the localization files, but kept in the general Roman calendar file.
> - `dedication_of_consecrated_churches` should be kept on the localization files.

Okay, I’ll remove the `dedication_of_consecrated_churches_on_*` l10n keys. Thanks! :pray:

**Comment by emagnier** [comment]
Indeed, you are correct. These celebrations are well indicated in the common of the missal (both in Latin and in vernacular). However, they are not mentioned in the GRC.

It is only suggested for specific countries/diocese on particular dates.

In the third and latest edition of the missal in French (2021), for France, Belgium, and Luxembourg, this celebration is indeed proposed on two possible dates: October 25th or the last Sunday of October.

<details>
<summary>For example, click here to see an extract from the proper of France.</summary>
<img src="https://github.com/romcal/romcal/assets/1045997/96443ab0-d114-4ee0-a001-40f9887d35b6" alt="" />
</details>

Good news is that in the work I have done on the 'optimization-ex' branch, it is already possible to integrate these celebrations on a case-by-case basis into particular calendars, without having to redefine them from scratch.

And for your information, I have the latest editions of the missals in Latin and French (in print), as well as Italian (in PDF).

**Comment by tukusejssirs** [comment]
> In the third and latest edition of the missal in French (2021), for France, Belgium, and Luxembourg, this celebration is indeed proposed on two possible dates: October 25th or the last Sunday of October.

I presume the two possible dates are because the celebration could be transfered to (usually the nearest/following) Sunday. Yes, the previous Sunday is not an option, but in general, I believe the rules are not that strict. Even in our parish, when the patron saint feast fall on a weekday, it is actually celebrated solemnly during either the previous or the following weekend. I assume the same is the case even for churches with unknown consecration date in French-speaking countries (I can be wrong though).

→ Therefore, my suggestion for the `france*.ts` calendars is to use `dedication_of_consecrated_churches` and a single celebration definition per calendar with a n option in the celebration definition which would define whether it should be transferred to the last Sunday of October. YMMV.

---

> And for your information, I have the latest editions of the missals in Latin and French (in print), as well as Italian (in PDF).

Awesome! When you have time and wish to take some pictures of select pages, I want to make them the first-party sources of our l10n data in romcal. OFC, no hurry, as I am currently working on `hu.ts`, then I want to finally finish #393 (which I need to start again, beginning with taking the very same pictures of `mr_sk_2021_ed3`, as previously I based my work on _Ordo 2023_ which differs in some way from `mr_sk_2021_ed3`). That said, I wish you could contribute this publicly, so that I could publish it on gDrive.

Simply put: I have some much work and tasks (work, family, romcal; and even within romcal) that I need to prioritise and the tasks in order. Hopefully, I can devote more time to romcal in the upcoming months, as I have some ideas (even regarding the monorepo, but more on that later).

**Comment by emagnier** [comment]
I'm on board with keeping only one of the two definitions of celebrations in the particular calendars I mentioned above. Then, it's up to a sub proper calendar (under `france-*`) to transfer or remove that date if necessary.

Following this logic through, we should remove this celebration from the GRC.

**Comment by tukusejssirs** [comment]
> Following this logic through, we should remove this celebration from the GRC.

I believe that is the correct way, as the pure GRC is nowhere used, it is just a base calendar from which derive all the others.
